build/deepcopy.bzl

```diff
@@ -0,0 +1,52 @@
+_cmd = """
+  # location of prebuilt deepcopy generator
+  dcg=$$PWD/'$(location //vendor/k8s.io/code-generator/cmd/deepcopy-gen)'
+  # original pwd
+  export O=$$PWD
+  # gopath/goroot for genrule
+  export GOPATH=$$PWD/.go
```

Comment:
the `go_genrule` rule from repo-infra probably does what you want here.

Reply:
I'm not sure what it's doing incorrectly with staging directories, but probably.

The other alternative is using the `go_path` rule from `rules_go`, which does similar things to what you're trying to do here. I had it mostly working with our existing bazel code generators (as an alternative to `go_genrule`) though there were still a few lingering issues. (I'm also not sure where that commit snuck off to.)

Reply:
We need to ignore `$GOPATH/src/k8s.io/kubernetes/staging/src/FOO/` and instead look in `$GOPATH/src/k8s.io/kubernetes/vendor/FOO` instead

Reply:
`go_genrule` just creates a new go workspace and symlinks to the original tree, similar to what `hack/run-in-gopath.sh` does (and which is what make does, I think).

does deepcopy not read through the vendor symlinks?

Reply:
Thanks for the tip!

Refactored to use `go_genrule`